src/Compilers/CSharp/Portable/Symbols/RangeVariableSymbol.cs

```diff
@@ -164,15 +162,21 @@ public override bool Equals(Symbol obj, TypeCompareKind compareKind)
                 return true;
             }
 
-            var symbol = obj as RangeVariableSymbol;
-            return (object)symbol != null
-                && symbol._locations[0].Equals(_locations[0])
+            // If we have no location, we have no way to compare two distinct range variables for equality.  So if we
+            // don't have the exact same instance, we have to presume these are not the same.
+            if (_location is null)
+            {
+                return false;
+            }
```

Comment:
@AlekseyTs def not sure how best to handle this.  it would have crashed before anyways... so this doesn't appear to naturally be hittable?  Should we have this code, or should we instead assert?